lib/segment/src/types.rs

```diff
@@ -322,14 +323,15 @@ impl TryFrom<GeoPointShadow> for GeoPoint {
 }
 
 /// All possible payload types
-#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone)]
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone, PartialEq)]
 #[serde(rename_all = "snake_case")]
 #[serde(tag = "type", content = "value")]
 pub enum PayloadType {
     Keyword(Vec<String>),
     Integer(Vec<IntPayloadType>),
     Float(Vec<FloatPayloadType>),
     Geo(Vec<GeoPoint>),
+    Json(serde_json::Value),
```

Comment:
I was thinking more in the direction of replacing whole stored payload with just `serde_json::Value` and only keep types in the index. The reason for this is that we currently do not preserve the difference between
```
{
  "key": "val"
}
```

and 

```
{
  "key": ["val"]
}
```

Reply:
I extended the `PayloadType` object because this approach has two advantages:

1. The new payload can still deserialize the old format. We keep the storage compatible.
2. Play well with the "strict" API .

But, if those are not concerns, we can go ahead and replace the  whole payload structure.